packages/client/src/runtime/core/extension/$extends.ts

```diff
@@ -0,0 +1,70 @@
+import { Required } from 'ts-toolbelt/out/Object/Required'
+import { Undefinable } from 'ts-toolbelt/out/Object/Undefinable'
+
+import { actionOperationMap, Client } from '../../getPrismaClient'
+import { createRequestProxy, RequestProxyOptions } from '../request/createRequestProxy'
+
+export type ExtensionOptions = {
+  type: string
+} & ResultOptions &
+  ModelOptions &
+  ClientOptions &
+  QueryOptions
+
+type ResultOptions = {
+  result: {
+    needs: {
+      [K in string]: boolean | ResultOptions['result']['needs']
+    }
+    fields: {
+      [K in string]: () => unknown | Promise<unknown>
+    }
+  }
+}
+
+type ModelOptions = {
+  model: {
+    [K in string]: () => unknown | Promise<unknown>
+  }
+}
+
+type ClientOptions = {
+  model: {
+    [K in string]: () => unknown | Promise<unknown>
+  }
+}
+
+type QueryOptions = {
+  query: {
+    [key in keyof typeof actionOperationMap]: () => unknown | Promise<unknown>
+  } & {
+    $nested: {
+      create: {
+        [K in string]: {} | null | undefined | QueryOptions['query']['$nested']['create']
+      }
+      update: {
+        [K in string]: {} | null | undefined | QueryOptions['query']['$nested']['update']
+      }
+      upsert: {
+        [K in string]: {} | null | undefined | QueryOptions['query']['$nested']['upsert']
+      }
+      where: {
+        [K in string]: {} | null | undefined | QueryOptions['query']['$nested']['where']
+      }
+    }
+  }
+}
+
+/**
+ *
+ * @param this
+ */
+export function $extends(this: Client, input: any, proxyOptions: RequestProxyOptions = {}) {
+  const forkedProxyOptions: Undefinable<Required<RequestProxyOptions>> = {
+    forbiddenProps: proxyOptions.forbiddenProps ? [...proxyOptions.forbiddenProps] : undefined,
+    transaction: proxyOptions.transaction ? { ...proxyOptions.transaction } : undefined,
+    extensions: proxyOptions.extensions ? [...proxyOptions.extensions, input] : undefined,
```

Comment:
```suggestion
    extensions: (proxyOptions.extensions ?? []).concat(input)
```

Shouldn't we always add `input` to extensions, even they were undefined initially?